docfx_yaml/extension.py

```diff
@@ -517,22 +517,51 @@ def _update_friendly_package_name(path):
+            type_map = {}
+            if argspec.annotations:
+                for annotation in argspec.annotations:
+                    if annotation == "return":
+                        continue
+                    # Extract names for simple types.
+                    try:
+                        type_map[annotation] = (argspec.annotations[annotation]).__name__
+                    # Try to extract names for more complicated types.
+                    except AttributeError:
+                        vartype = argspec.annotations[annotation]
+                        try:
+                            type_map[annotation] = str(vartype._name)
+                            if vartype.__args__:
+                                type_map[annotation] += str(vartype.__args__)[:-2] + ")"
+                        except AttributeError:
+                            print(f"Could not parse argument information for {annotation}.")
+                            continue
+
             for arg in argspec.args:
+                arg_map = {}
                 # Ignore adding in entry for "self"
                 if arg != 'cls':
-                    args.append({'id': arg})
+                    arg_map['id'] = arg
+                    if arg in type_map:
+                        arg_map['var_type'] = type_map[arg]
+                        args.append(arg_map)
             if argspec.varargs:
                 args.append({'id': argspec.varargs})
             if argspec.varkw:
                 args.append({'id': argspec.varkw})
+            # Try to add default values. Currently does not work if there is * argument present.
             if argspec.defaults:
-                for count, default in enumerate(argspec.defaults):
-                    cut_count = len(argspec.defaults)
-                    # Only add defaultValue when str(default) doesn't contain object address string(object at 0x)
-                    # inspect.getargspec method will return wrong defaults which contain object address for some default values, like sys.stdout
-                    # Match the defaults with the count
-                    if 'object at 0x' not in str(default):
-                        args[len(args) - cut_count + count]['defaultValue'] = str(default)
+                # Attempt to add default values to arguments.
+                try:
+                    for count, default in enumerate(argspec.defaults):
+                        cut_count = len(argspec.defaults)
+                        # Only add defaultValue when str(default) doesn't contain object address string(object at 0x)
+                        # inspect.getargspec method will return wrong defaults which contain object address for some default values, like sys.stdout
+                        # Match the defaults with the count
+                        if 'object at 0x' not in str(default):
+                            args[len(args) - cut_count + count]['defaultValue'] = str(default)
```

Comment:
I don't understand `len(args) - cut_count + count`. Can you clarify in a comment, different variable names, or something else?